Memory test on scrolling large images quickly

liyuqian opened the pull request:
For https://github.com/flutter/flutter/issues/19558

This test reveals the issue that Flutter currently retains as much as
200MB memory after scrolling. (200MB can further grow if we scroll more
times). There should be plenty of time for the GC to clean such memory.

liyuqian committed:
Memory test on scrolling large images quickly

For https://github.com/flutter/flutter/issues/19558

This test reveals the issue that Flutter currently retains as much as
200MB memory after scrolling. (200MB can further grow if we scroll more
times). There should be plenty of time for the GC to clean such memory.

dnfield commented on dev/benchmarks/macrobenchmarks/lib/src/large_images.dart:
```diff
@@ -0,0 +1,25 @@
+      gridDelegate:
+      const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 3),
+      itemBuilder: (BuildContext context, int index) {
+        return Image.network(
```
Can we avoid using a network image for this?  If the server goes down our test will fail right?

liyuqian replied:
I tried using local images in https://github.com/flutter/flutter/compare/master...liyuqian:local_image_test but unfortunately it no longer reproduces the memory issue. It seems that some image downloading or decompressing is involved in this memory issue.

I just strengthened the test to scroll even faster so we can see a 500MB+ (sometimes 700MB) memory usage. One more scroll crashed the test in my Moto G4.

dnfield replied:
What if we do it with a asset image that isn't cached so we have to hit the disk each time we access it?

Or what if we made an image provider that simulates a delay as if it was coming from the network each time?

If we _really_ need to hit the network for this, we should probably see if we can hit a domain that we're more confident in or need to have up anyway (like some image from pub.dev or flutter.dev).

liyuqian committed:
Increase scrolling speed

Now we'll see 500MB+ (sometimes 700MB) memory usages.

One more scroll may crash the app on Moto G4.

liyuqian committed:
Correct license headers

liyuqian commented:
Closing this in favor of https://github.com/flutter/flutter/pull/46184
